### PR TITLE
PRL-4616:added changes for sendgrid templates

### PR DESCRIPTION
**Pull request by khurshedk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-4616

### Change description ###

Sendgrid email template changes for Cafcass Cymru

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/models/email/SendgridEmailTemplateNames.java [hunk 1]
```diff
@@ -2,5 +2,7 @@
 
 public enum SendgridEmailTemplateNames {
 
-    SERVE_ORDER_ANOTHER_ORGANISATION
+    SERVE_ORDER_ANOTHER_ORGANISATION,
+    SERVER_ORDER_CAFCASS_CYMRU
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/models/email/SendgridEmailTemplateNames.java:6]: SERVER?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java [hunk 2]
```diff
@@ -389,12 +390,8 @@ public void sendEmailToCafcassAndOtherParties(CaseDetails caseDetails) {
...
         cafcassEmails.addAll(otherEmails);
 
-        cafcassEmails.forEach(email -> emailService.send(
-            email,
-            EmailTemplateNames.CAFCASS_OTHER,
-            buildEmailToCafcassAndOtherParties(caseData),
-            LanguagePreference.english
-        ));
+        List<Document> orderDocuments = getServedOrderDocumentsAndAdditionalDocuments(caseData);
+        sendEmailToCafcassCymru(caseData,cafcassEmails,systemUserService.getSysUserToken(),orderDocuments);
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:394]: Is this method called at all?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java [hunk 3]
```diff
@@ -479,20 +476,55 @@ public void sendEmailWhenOrderIsServed(String authorisation,
...
         // Send email notification to other organisations
-        listOfOtherAndCafcassEmails.forEach(email ->
-                                                emailService.send(
-                                                    email,
-                                                    EmailTemplateNames.CAFCASS_OTHER,
-                                                    buildEmailToCafcassAndOtherParties(caseData),
-                                                    LanguagePreference.english
-                                                )
-        );
+        sendEmailToCafcassCymru(caseData,listOfOtherAndCafcassEmails,authorisation, orderDocuments);
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:479]: This call needs to be moved inside C100 check. and the comment above method call is not correct

**Review by shashisk97 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java [hunk 4]
```diff
@@ -448,30 +417,68 @@ public void sendEmailWhenOrderIsServed(String authorisation,
             //PRL-4225 - set bulkIds in the orderCollection & update in caseDataMap
             addBulkPrintIdsInOrderCollection(caseData, bulkPrintOrderDetails);
             caseDataMap.put(ORDER_COLLECTION, caseData.getOrderCollection());
+            sendEmailToCafcassCymru(caseData,listOfOtherAndCafcassEmails,authorisation, orderDocuments);
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:420]: are we sending email to anyother party than cafcass? if not why the name is listOfOtherAndCafcassEmails?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java [hunk 5]
```diff
@@ -448,30 +417,68 @@ public void sendEmailWhenOrderIsServed(String authorisation,
...
-                                                )
-        );
+    }
 
+    private void sendEmailToCafcassCymru(CaseData caseData, List<String> cafcassEmailInformation,
+                                        String authorisation, List<Document> orderDocuments) {
+
+        Map<String, Object> dynamicData = getDynamicDataForEmail(caseData);
+        dynamicData.put("dashBoardLink",manageCaseUrl + "/" + caseData.getId() + ORDERS);
+        cafcassEmailInformation.stream().forEach(emailAddress -> {
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:442]: Why is this a list, cafcass cymru will have only one email id

File: src/functionalTest/resources/requests/cafcass-cymru-send-email-request.json [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+        ],
+        "list_items": [
+          {
+            "code": "6bb5e9ac-df97-4593-8b22-3969dc0bb4e1",
+            "label": "Sam Nolan"
+          }
+        ]
+      },
+      "cafcassCymruServedOptions": "Yes",
+      "cafcassCymruEmail": "khurshed.khan@hmcts.net",
```
Comment (shashisk97) [inline src/functionalTest/resources/requests/cafcass-cymru-send-email-request.json:299]: PLease remove this hmcts email id.

Comment (khurshedk) [reply src/functionalTest/resources/requests/cafcass-cymru-send-email-request.json]: @shashisk97 as discussed removed mentioned email id

**Review by shashisk97 — APPROVED** [review]

**Review by yogendraU — APPROVED** [review]

**Pull request merged**